packages/flutter_tools/test/general.shard/dart_plugin_test.dart

```diff
@@ -994,8 +1081,11 @@ void createFakeDartPlugins(
 ) {
   final Directory fakePubCache = fs.systemTempDirectory.childDirectory('cache');
   final File packagesFile = flutterProject.directory
-    .childFile('.packages')
-    ..createSync(recursive: true);
+    .childFile('.packages');
+  if (packagesFile.existsSync()) {
+    packagesFile.deleteSync();
+  }
```

Comment:
One test re-runs `createFakeDartPlugins`, and the fact that this clearing step wasn't necessary was a side-effect of the fact that the previous implementation would not create registration when the only plugin was an implementation for which neither the app-facing package nor a direct dependency was present. That logic changed, which broke the test even though that wasn't the point of that test.